docs/api/utils.rst

```diff
@@ -0,0 +1,604 @@
+
+-----
+
+   .. method:: CalcImageMatrix(int width, int height, Rect tr, float rotate, bool keep)
+
+      Compute image insertion matrix
+
+      :arg int width: image width
+      :arg int height: image height
+      :arg Rect tr: rect of target image
+      :arg float rotate: rotate to be set for target image
+      :arg bool keep: calc size of target image keeping origin image's ratio
+      
+      :returns: Matrix mat
+
+-----
+
+   .. method:: ColorCount(FzPixmap pm, dynamic clip)
+
+      Return count of each color.
+
+      :arg FzPixmap pm: source pixmap
+      :arg Rect clip: count colors in clip area of source pixmaps.
+
+      :returns: return count of each color.
+
+-----
+
+   .. method:: ConstructLabel(string style, string prefix, int pno)
+
+      Construct a label based on style, prefix and page number.
+
+      :arg string style: type of style for label. that includes `D`, `r`, `R`, `a`, `A`.
+      :arg string prefix: added prefix to label
+      :arg int pno: translate pno to letter according to the style
+
+      :rtype: string
+      :returns: styled label
+
+-----
+
+   .. method:: DecodeRawUnicodeEscape(FzBuffer s)
+   .. method:: DecodeRawUnicodeEscape(string s)
+
+      Decode raw unicode
+
+-----
+
+   .. method:: DoLinks(Document doc1, Document doc2, int fromPage: -1, int toPage: -1, int startAt: -1)
+
+      Insert links contained in copied page range into destination PDF.
+
+-----
+
+   .. method:: EnsureIdentity(Document pdf)
+
+      Store ID in PDF trailer
+
+-----
+
+   .. method:: ExpandFontName(string fontname)
+
+      Make /DA string of annotation
+
+      :returns: expand font name. For example, if fontname starts with `co` or `Co`, returns `Cour`
+
+-----
+
+   .. method:: GetId()
+
+      Count numbers and return unique id on one process
+
+      :rtype: int
+      :returns: unique number
+
+-----
+
+   .. method:: GetAllContents(Page page)
+
+      All /Contents streams concatenated to one bytes object.
+
+      :arg Page page: Page object to get all streams
+
+      :rtype: bytre[]
+
+-----
+
+   .. method:: GetAnnotByName(Page page, string name)
+
+      Retrieve annot by name (/NM key)
+
+      :arg Page page: Page object containing annots.
+      :arg string name: annot name. Looping annots in page, that find the annot which has name as ID.
+
+      :rtype: PdfAnnot
+      :returns: PdfAnnot object that has the name.
+
+-----
+
+   .. method:: GetArea(Rect rect, string unit: "px")
+
+      Calculate area of rectangle.\nparameter is one of 'px' (default), 'in', 'cm', or 'mm'.
+
+      :arg Rect rect: rectangle calculated area
+      :arg string unit: unit used in rect. default is `px`, there are other units like `cm`, `mm`.
+
+      :rtype: float
+      :returns: area of rectangle
+
```

Comment:
Everything else below here - if it is a datatype please codify it as such:

i.e 
If this is a C# object then can we codify it please? ie. int - > `int`. Or maybe it is `Int`? etc. 
float[] -> `float[]`